File: C++/DayOne.cpp
#include "Days.h"

int Captcha(const std::string Input)
{
    int Output = 0;

    for(int i = 0; i < Input.length(); i++)
    {
        if(i == Input.length() - 1)
        {
            if (Input[i] == Input[0])
                Output += stoi(string(1, Input[i]));

            break;
        }

        if(Input[i] == Input[i + 1])
            Output += stoi(string(1, Input[i]));
    }

    return Output;
}

int CaptchaPtTwo(const std::string Input)
{
    int Output = 0;
    const int Offset = Input.length() / 2;

    for(int i = 0; i < Input.length(); i++)
    {
        int Index = i + Offset;

        if(Index >= Input.length())
            Index = Index - Input.length();

        if(Input[i] == Input[Index])
            Output += stoi(string(1, Input[i]));
    }

    return Output;
}

File: Rust/day_one.rs
pub fn captcha(input: &str) -> u32 {
    let mut output = 0;
    
    for i in 0..input.len() {
        let character = input.chars().nth(i).unwrap();

        if i == input.len() - 1 {
            if character == input.chars().nth(0).unwrap() {
                output += character.to_digit(10).unwrap();
            }
            break;
        }

        if character == input.chars().nth(i + 1).unwrap() {
            output += character.to_digit(10).unwrap();
        }
    }

    return output;
}

pub fn captcha_pt_two(input: &str) -> u32 {
    let mut output = 0;
    let offset = input.len() / 2;

    for i in 0..input.len() {
        let character = input.chars().nth(i).unwrap();
        let mut next_index = i + offset;

        if next_index >= input.len() {
            next_index = next_index - input.len()
        }

        if character == input.chars().nth(next_index).unwrap() {
            output += character.to_digit(10).unwrap();
        }
    }

    return output;
}